algebird-test/src/main/scala/com/twitter/algebird/BaseProperties.scala

```diff
@@ -27,8 +27,8 @@ import scala.math.Equiv
 object BaseProperties extends MetricProperties {
   val arbReasonableBigDecimals: Arbitrary[BigDecimal] = Arbitrary(
     for {
-      scale <- Gen.choose(-128, +128)
-      base <- implicitly[Arbitrary[BigInt]].arbitrary
+      scale <- Gen.choose(-4, +4)
+      base <- implicitly[Arbitrary[Int]].arbitrary
```

Comment:
@johnynek: This may not be a good idea. I can go into it but when you add two BigDecimals that have sufficiently different scales it is possible to lose precision. And when testing out the distributive properties of rings the way in which the precision is lost is not distributive. I suspect nobody is surprised by this though so it's reasonable to limit the tests to cases where this does not happen.

Reply:
This is fine, but can we do the biggest scale we can do with the tests passing consistently? Would 24 work for instance?